dist/init/linux-systemd/caddy.service

```diff
@@ -5,18 +5,32 @@ After=network-online.target
 Wants=network-online.target systemd-networkd-wait-online.service
 
 [Service]
+; Only allow 1 restart with 10 seconds so that caddy doesn't get locked out by PAM
 Restart=on-failure
-StartLimitInterval=86400
-StartLimitBurst=5
+StartLimitInterval=10
+StartLimitBurst=1
```

Comment:
> With systemd you can tell startup phase (»starting«) from a runtime phase (»serving«) and therefore kill a process if it didn't start within some seconds, or hangs.

That's a nifty feature. Caddy may take various amounts of time to start up though, with cert management taking place.